### PR TITLE
check for a time.ParseError, add unit tests

### PR DESCRIPTION
**Pull request by bacongobbler**
refs #10

### CODE REVIEW

File: time/time.go [hunk 1]
```diff
@@ -3,7 +3,8 @@ package time
 import "time"
 
 // DeisDatetimeFormat is the standard date/time representation used in Deis.
-const DeisDatetimeFormat  string = "2006-01-02T15:04:05MST"
+const DeisDatetimeFormat string = "2006-01-02T15:04:05MST"
```
Comment (arschles) [inline time/time.go]: don't need the `string` after the const name here. same with the below

File: time/time.go [hunk 2]
```diff
@@ -27,7 +28,7 @@ func (t *Time) MarshalJSON() ([]byte, error) {
 // The time is expected to be in Deis' datetime format.
 func (t *Time) UnmarshalText(data []byte) (err error) {
```
Comment (arschles) [inline time/time.go]: can you remove the named return param here? it's shadowed below and hence not used

File: time/time.go [hunk 3]
```diff
@@ -27,7 +28,7 @@ func (t *Time) MarshalJSON() ([]byte, error) {
...
 	tt, err := time.Parse(DeisDatetimeFormat, string(data))
-	if err != nil {
+	if _, ok := err.(*time.ParseError); ok {
 		tt, err = time.Parse(PyOpenSSLTimeDateTimeFormat, string(data))
 	}
```
Comment (arschles) [inline time/time.go:33]: what about other errors? I'd like to be defensive here in case other errors ever are returned

File: time/time.go (hunk 1 above)
Comment (bacongobbler) [inline time/time.go]: good points. Old code but isn't necessary yeah.

File: time/time.go (hunk 3 above)
Comment (arschles) [inline time/time.go:33]: same with the below

Comment (bacongobbler) [inline time/time.go:33]: I've refactored the function to explicitly return the error, which was the original intention. Does that look better to you?

Comment (arschles) [inline time/time.go:33]: yup! and thanks for removing the named return param too

File: time/time_test.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+func TestUnMarshalText(t *testing.T) {
+	dummyTime := Time{}
+
+	goodTimeFormats := []string{
+		"2006-01-02T15:04:05MST",
+		"2006-01-02T15:04:05",
+	}
+	for _, goodTime := range goodTimeFormats {
+		if dummyTime.UnmarshalText([]byte(goodTime)) != nil {
```
Comment (arschles) [inline time/time_test.go:17]: for some extra sanity, can you split the formats and actual values apart, so the tests ensures that multiple times of the given format work?

**Comment by arschles** [comment]
@bacongobbler left one final comment regarding tests. LGTM after that's addressed

File: time/time_test.go (hunk 4 above)
Comment (bacongobbler) [inline time/time_test.go:17]: certainly

**Comment by bacongobbler** [comment]
Okay I split them out by time format. I also embedded the time struct so you no longer have to call `t.Time.Year()` every time you want to check the year of the object. Things like `time.Now()` still won't be available, but it should make the deis-specific time struct much more usable. I added some tests checking the Year to demonstrate its usage. It's also completely backwards compatible because you can still call `t.Time` to get the raw Time struct.

**Comment by arschles** [comment]
:+1:

**Comment by bacongobbler** [comment]
I will admit that this has gone out of scope of the original intention, so please let me know if you'd be happier with this all split out into different PRs.

**Comment by arschles** [comment]
@bacongobbler I'm good with it as-is because you've added tests :)

**Comment by bacongobbler** [comment]
tests make everyone happy <3

**Pull request merged**